core/player/src/view/parser/utils.ts

```diff
@@ -1,8 +1,28 @@
 import type { Node } from "./types";
 
-/** Check to see if the object contains async */
-export function hasAsync(obj: object): boolean {
-  return Object.prototype.hasOwnProperty.call(obj, "async");
+/**
+ * Checks if there are templated values in the object
+ *
+ * @param obj - The Parsed Object to check to see if we have a template array type for
+ * @param localKey - The key being checked
+ */
+export function hasTemplateValues(obj: any, localKey: string) {
+  return (
+    Object.hasOwnProperty.call(obj, "template") &&
+    Array.isArray(obj?.template) &&
+    obj.template.length &&
+    obj.template.find((tmpl: any) => tmpl.output === localKey)
+  );
+}
+
+/** Check to see if the string is a valid switch key */
+export function hasSwitchKey(localKey: string) {
+  return localKey === ("staticSwitch" || "dynamicSwitch");
```

Comment:
Huh, I didn't know this syntax was possible to check if a string was either value.

Reply:
i would have assumed `("staticSwitch" || "dynamicSwitch")` would have evaluated to `true` so it would have just checked if `localKey` was `true`

Reply:
So seems like this syntax does not work. This was pre-existing code that I moved over to `utils.ts` from parser `index.ts`. I'll leave the fix for this out of this PR and defer it to another ticket for now!

Reply:
Since we're making changes in the area I would opt to just make the changes now. Its also concerning that we don't have a test that would have caught this?

Reply:
I made #453 to track this, to make sure we're capturing additional work added to this long running task